wip: blog pattern

jmuzina opened the pull request:
## Done

Builds the [blog pattern](https://www.figma.com/design/1brHCsyeTE6FU7hfJd6ao2/%F0%9F%9A%A7---WIP--Sites---Core-component-library?node-id=3168-31983&m=dev)

## QA

- Open [demo](insert-demo-url)
- [Add QA steps]
- Review updated documentation:
  - [List any updated documentation for review]

### Check if PR is ready for release

If this PR contains Vanilla SCSS or macro code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention
  - if existing APIs (CSS classes & macro APIs) are not changed it can be a bugfix release (x.x.**X**)
  - if existing APIs (CSS classes & macro APIs) are changed/added/removed it should be a minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) or macros should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://vanilla-framework-5659.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/26044)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

eliman11 commented:
Thanks @jmuzina and @kim-isaac! Some minor comments - 
- [x] Doesn't affect the previews as we're using the muted heading style anyway, but could we change "Latest from our Blog" to use sentence case (i.e. "Latest from our blog")
- [ ] Since these blog sections are usually standalone sections with the same hierarchy as other H2 headings on the page, could we use the muted heading style but use the H2 label instead of the H4? More so for users using screen readers to ensure we aren't skipping heading hierarchies
- [ ] Similarly the article headings should then take on H3 labels instead of H5s
- [ ] Maybe more of a question for @kim-isaac but should we allow for some articles within the same block to show metadata, and some articles without metadata? Or even if it is possible with the pattern, would we want to showcase this as the first example?
<img width="1068" height="371" alt="Screenshot 2025-10-23 at 17 54 26" src="https://github.com/user-attachments/assets/d90f15d3-9aff-4678-b589-729c92f64415" />

jmuzina commented:
Thanks @eliman11 I addressed all your points except for the question for @kim-isaac about mixed metadata presence, can you have another look?

Edit: It may take a bit for the demo to update, it looks like there is some slowness going on on the infra side.

jmuzina commented on templates/_macros/shared/vf_article_block.jinja:
```diff
@@ -0,0 +1,187 @@
+
+        <div class="p-article-block__item">
+            {{- _article_metadata(authors=authors, date=date, template_mode=template_mode) -}}
+        </div>
```
See https://github.com/canonical/vanilla-framework/pull/5659#discussion_r2457425023

kim-isaac commented:
Thanks @jmuzina ! Here are a few more comments:

- [ ] Regarding @eliman11 's question, I agree that the metadata should either appear consistently or not appear at all within a single section.
- [ ] In the document, please use the default title style (not the link style) as this would be the standard setting.
<img width="414" height="375" alt="image" src="https://github.com/user-attachments/assets/25ec9066-be47-4755-82a6-94d8a2460093" />

- [ ] It seems both p-article-block__description article-excerpt and the nested p have 6px top padding, causing overlap - could we remove one of them?
- [ ] Could we add 8px bottom spacing between stacked blocks in medium and small dimensions to match the current equal-heights row?

kim-isaac commented:
Thanks @jmuzina , looks great! I'll add +1

muhammad-ali-pk commented:
Minor:

There are empty spaces in some classes, and some indentation issues
<img width="765" height="119" alt="image" src="https://github.com/user-attachments/assets/b8609b80-7c16-458c-8b0a-0a9009ad3eeb" />
